### PR TITLE
[LLM] llm transformers api support batch actions

### PR DESCRIPTION
**Pull request by cyita**
## Description

Support batch functions in transformers api.

### 1. Why the change?

Align with transformers' API.

#### Use bigdl's tokenizer
```python
llm = Gptneox("/d1/llm/models/nano-gptneox-7b-redpajama-q4_0.bin",
              use_mmap=False, n_threads=28, n_batch=8)
prompt = "Q: Tell me something about Intel. A:"
tokens = llm.tokenize([prompt, prompt])
tokens_id = llm.generate(tokens, max_new_tokens=32)
decoded = llm.batch_decode(tokens_id)
print(decoded)
```

#### Use transformers' tokenizer
```python
tokenizer = AutoTokenizer.from_pretrained(model_id)
llm = Gptneox("/d1/llm/models/nano-gptneox-7b-redpajama-q4_0.bin",
              use_mmap=False, n_threads=28, n_batch=8)
tokens = tokenizer([prompt, prompt]).input_ids
tokens_id = llm.generate(tokens, max_new_tokens=32)
print(tokenizer.batch_decode(tokens_id))
```

### CODE REVIEW

**Review by TheaperDeng — APPROVED** [review]
please add numpy to required dependencies

File: python/llm/src/bigdl/llm/ggml/model/generation/utils.py [hunk 1]
```diff
@@ -30,33 +31,64 @@ class GenerationMixin:
...
+                bstr = t
+            result.append(self._tokenize(bstr, add_bos))
 
-    def decode(self, tokens: List[int]) -> str:
+        if not is_batched:
+            result = result[0]
+        return result
+
+    def decode(self,
+               tokens: Union[List[int], List[List[int]], "np.ndarray"]) -> str:
```
Comment (TheaperDeng) [inline python/llm/src/bigdl/llm/ggml/model/generation/utils.py:62]: should we rename this to `batch_decode`?

Comment (cyita) [reply python/llm/src/bigdl/llm/ggml/model/generation/utils.py:62]: Yes, `batch_decode` is better.

**Comment by cyita** [comment]
> please add numpy to required dependencies

Already in CONVERT_DEP.

**Pull request merged**